src/java.base/share/classes/java/net/URLConnection.java

```diff
@@ -1133,6 +1141,10 @@ public void setRequestProperty(String key, String value) {
      * key-value pair.  This method will not overwrite
      * existing values associated with the same key.
      *
+     * This method could be a no-op if appending a value
+     * to the map is not supported by the protocol being
+     * used in a given subclass.
+     *
```

Comment:
I'm not sure about this. What is the use-case driving this clarification? Is it just for the corner case where the default implementation of getHeaderFields is not overridden ?

Reply:
Many protocol handlers don't support adding request headers (ftp://, file://, mailto:// jrt://) - they all inherit form s.n.www.URLConnection which has this implementation:

```
    public void addRequestProperty(String key, String value) {
        if (connected)
            throw new IllegalStateException("Already connected");
        if (key == null)
            throw new NullPointerException ("key is null");
    }
```